src/mmw/js/src/modeling/models.js

```diff
@@ -270,16 +271,18 @@ var ScenarioModel = Backbone.Model.extend({
                     name: 'precipitation',
                     value: 1.0
                 }
-            ]
+            ],
+            // This is the md5 hash of the above default value
+            inputmod_hash: 'ab0d9b8a1d441b8ec7ddf81b5a49dd2d'
```

Comment:
This is a really funny problem to tackle. I agree with what you're saying though: having a hard coded hash is pretty brittle. The reason we set it here is that in Backbone the JSON string representation of the inputs is this:

```
[{"name":"precipitation","value":1.8,"type":"","shape":null,"area":"0","units":"m<sup>2</sup>"}]
```

whereas the value that is initialized from the server is this:

```
[{"name":"precipitation","area":"0","value":1.8,"shape":null,"units":"m<sup>2</sup>","type":""}]
```

Identical objects, but different string representations, leading to different MD5 hashes. Thus, when loading an existing project, a client-side generated MD5 hash (of the server representation) would not match the saved MD5 hash (of the client representation) from the server, leading to a redundant model run when it is not required.

I'm looking in to what can be done about it.